pueue/src/daemon/service.rs

```diff
@@ -0,0 +1,639 @@
+    fn stop(&self) {
+        let mut child = self.child.lock().unwrap();
+
+        self.request_stop.store(true, Ordering::Relaxed);
+        match child.kill() {
+            Ok(_) => {
+                debug!("stop() kill");
+                self.running.store(false, Ordering::Relaxed);
+                // signal the wait() to exit so a `while` condition is checked at least once more.
+                // as long as `while` conditions have been changed _before_ the call to stop(),
+                // the changed condition will be observed
+                _ = self.wait_tx.send(());
+            }
+
+            Err(e) => {
+                self.running.store(false, Ordering::Relaxed);
+                error!("failed to stop(): {e}");
+            }
+        }
+    }
+
+    /// wait for child process to exit
+    fn wait(&self) {
+        _ = self.wait_rx.lock().unwrap().recv();
+    }
+
+    /// did the spawned process quit without our request?
+    fn dirty(&self) -> bool {
+        self.dirty.load(Ordering::Relaxed)
+    }
+
+    /// try to spawn a child daemon
+    fn start(&self, session: Option<u32>) -> Result<()> {
+        let Some(session) = session.or_else(get_current_session) else {
+            bail!("get_current_session failed");
+        };
+
+        let running = self.running.clone();
+        let child = self.child.clone();
+        let waiter = self.wait_tx.clone();
+        let dirty = self.dirty.clone();
+        let request_stop = self.request_stop.clone();
+        _ = thread::spawn(move || {
+            request_stop.store(false, Ordering::Relaxed);
+
+            let res = run_as(session, move |token| {
+                let mut arguments = Vec::new();
+
+                let config = CONFIG
+                    .get()
+                    .ok_or_else(|| anyhow!("failed to get CONFIG"))?
+                    .clone();
+
+                if let Some(config) = &config.config_path {
+                    arguments.push("--config".to_string());
+                    arguments.push(format!(r#""{}""#, config.to_string_lossy().into_owned()));
+                }
+
+                if let Some(profile) = &config.profile {
+                    arguments.push("--profile".to_string());
+                    arguments.push(format!(r#""{profile}""#));
+                }
+
+                let arguments = arguments.join(" ");
+
+                // Try to get the path to the current binary
+                let current_exe = env::current_exe()?.to_string_lossy().to_string();
+
+                let mut command = format!(r#""{current_exe}" {arguments}"#)
+                    .encode_utf16()
+                    .chain(iter::once(0))
+                    .collect::<Vec<_>>();
+
+                // lpcommandline may modify the the cmd vec. max valid size is 1024, so
+                // I think it best to ensure 1024 bytes of capacity total exist just in case
+                command.reserve(1024 - (command.len() / 2));
```

Comment:
`lpcommandline` is the third argument of [`CreateProcessAsUserW`](https://learn.microsoft.com/en-us/windows/win32/api/processthreadsapi/nf-processthreadsapi-createprocessasuserw). From the docs it was not clear to me exactly _how much_ it was going to alter the string, so I put this here out of an abundance of caution to ensure no UB happens.

Reply:
According to the docs, 
> The system adds a null character to the command line string to separate the file name from the arguments. This divides the original string into two strings for internal processing. 

So 1 extra character may be enough. So I reduced this to 10 bytes instead (should be enough). Once I find clearer explanation than this (these docs are still a bit confusing) it could be fixed with 1 or whatever.

Reply:
Ok, got this resolved. It modifies the string _in-place_, so no buffer is needed. We can safely completely remove this portion of code.